src/frontegg-oauth-client.ts

```diff
@@ -155,15 +156,64 @@ const isTokenExpired = (tokenExpirationTime: number | null) => {
   return tokenExpirationTime - 60 * 60 * 1000 < Date.now()
 }
 
-const GET_FRONTEGG_USER_DATA_RESPONSE_SCHEMA = z.object({
-  id: z.string(),
+/**
+ * Function to decode a JWT token
+ *
+ * @param token the JWT token to decode
+ * @returns the decoded JWT token
+ */
+const decodeJwt = (token: string) => {
+  // Extract the base64 payload from the token
+  const base64Url = token.split('.')[1]
+  // Replace URL-safe characters with the base64 standard characters
+  let base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/')
+  // Add padding to the base64 string if needed, as base64 strings need to be a multiple of 4 characters
+  switch (base64.length % 4) {
+    case 0:
+      break
+    case 2:
+      base64 += '=='
+      break
+    case 3:
+      base64 += '='
+      break
+    default:
+      throw new Error('base64 string is not of the correct length')
+  }
+  // Decode the base64 string and replace special characters with their hex representation
+  // Only using atob would not work in all cases, as it does not decode all special characters
+  // See https://stackoverflow.com/a/30106551/10876985 for more details
+  const jsonPayload = decodeURIComponent(
+    atob(base64).replace(/(.)/g, (_m, p) => {
+      let code = (p as string).charCodeAt(0).toString(16).toUpperCase()
+      if (code.length < 2) {
+        code = `0${code}`
+      }
+      return `%${code}`
+    }),
+  )
+  // Parse the JSON payload and return it
+  return JSON.parse(jsonPayload)
+}
+
+const FRONTEGG_DECODED_TOKEN_SCHEMA = z.object({
+  sub: z.string().describe('JWT subject claim used for the Frontegg user id'),
   email: z.string(),
   name: z.string(),
   profilePictureUrl: z.string().nullable().optional(),
   tenantId: z.string(),
+  act: z
```

Comment:
🟡 It would be good to add some link to Frontegg docs for this variable. For someone like me (who sees this first time), it would be very helpful.

Reply:
Unfortunately there are none.